src/Servers/ShellServer.cs

```
@@ -146,6 +147,10 @@ private void EventLoop(NetMQSocket socket)
                             break;
                     }
                 }
+                catch (ProtocolViolationException ex)
+                {
+                    logger.LogCritical(ex, $"Protocol violation when trying to receive next ZeroMQ message.");
+                }
                 catch (ThreadInterruptedException)
```

Should we a catch-all Exception? Just to make sure we don't die?